server/aesdsocket.c

```diff
@@ -282,30 +397,55 @@ int main(int argc, char **argv)
     }
     
     init_sigaction();
-    addr_len=sizeof(addr);
+    addr_len=sizeof(client_addr);
+    
+    //create timestamp thread
+    pthread_t timestamp_thread;
+    threadArgs *timer_thread_args=(threadArgs *)malloc(sizeof(threadArgs));
+    if(timer_thread_args==NULL)
+    {
+    	syslog(LOG_ERR,"memory allocation for timer thread arguments failed");
+    }
+    else
+    {
+    	timer_thread_args->data_fd=data_fd;
+    	if(pthread_create(&timestamp_thread,NULL,time_stamp,(void *)timer_thread_args) !=0)
+    	{
+    		syslog(LOG_ERR,"failed to create timer thread");
+    		free(timer_thread_args);
+    	}
+    }
     while (!caught_sig)
     {
         
-        fd = accept(sockfd, (struct sockaddr *)&addr, &addr_len);
+        fd = accept(sockfd, (struct sockaddr *)&client_addr, &addr_len);
         if (fd == -1)
         {
             syslog(LOG_ERR, "accept() failed: %s", strerror(errno));
             continue;
         }
-    	inet_ntop(addr.ss_family, &(((struct sockaddr_in*)&addr)->sin_addr), client_ip, sizeof(client_ip));
-        syslog(LOG_DEBUG, "accept() successful from %s", client_ip);
-        
-        if(rx_socket_data(fd,data_fd)==0)
-        {
-        	send_to_client(fd,data_fd);
-        }
-        if(close(fd)==-1)
-        {
-        	syslog(LOG_ERR,"closing connection failed");
-        }
+    	pthread_t thread_id;
+    	threadArgs *thread_args=(threadArgs *)malloc(sizeof(threadArgs));
+    	if(thread_args==NULL)
+    	{
+    		syslog(LOG_ERR,"memory allocation for thread arguments failed");
+    		close(data_fd);
+    		continue;
+    	}
+    	thread_args->fd=fd;
+    	thread_args->data_fd=data_fd;
+    	thread_args->addr=client_addr;
+    	if(pthread_create(&thread_id,NULL,thread_function,(void *)thread_args) !=0)
+    	{
+    		syslog(LOG_ERR,"failed to create timer thread");
+    		free(thread_args);
+    		close(data_fd);
+    		continue;
+    	}
+    	add_new_thread_node(thread_id);
         
      }
-     
+     wait_for_all_threads_to_join();
```

Comment:
@shruthi-thallapally what does it mean to do cleanup here if the program runs for a long time before exiting?